tests/_multi_screen.lua

```diff
@@ -0,0 +1,509 @@
+                align  = "center",
+                widget = wibox.widget.textbox,
+            },
+            bg                 = colors[i],
+            fg                 = "#ffffff",
+            shape_border_color = "#ff0000",
+            shape_border_width = 1,
+            shape              = shape.rectangle,
+            widget             = wibox.container.background
+        }
+        local sur = surface.widget_to_surface(
+            w,
+            s.geometry.width,
+            s.geometry.height
+        )
+        wallpaper.fit(sur, s)
+    end
+end
+
+local function add_steps(real_steps, new_steps)
+    for _, dispo in ipairs(dispositions) do
+        -- Cleanup
+        table.insert(real_steps, function()
+            if #client.get() == 0 then return true end
+
+            for _, c in ipairs(client.get()) do
+                c:kill()
+            end
+        end)
+
+        table.insert(real_steps, function()
+            clear_screen()
+            local keep = dispo.keep or false
+            local old = {}
+            local geos = {}
+
+            check_tag_indexes()
+
+            if keep then
+                for _, sf in ipairs(dispo) do
+                    local geo = sf and sf() or nil
+
+                    -- If the function return nothing, assume the screen need to
+                    -- be destroyed.
+                    table.insert(geos, geo or false)
+                end
+
+                -- Removed screens need to be explicit.
+                assert(#geos >= screen.count())
+
+                -- Keep a cache to avoid working with invalid data
+                local old_screens = {}
+                for s in screen do
+                    table.insert(old_screens, s)
+                end
+
+                for i, s in ipairs(old_screens) do
+                    -- Remove the screen (if no new geometry is given
+                    if not geos[i] then
+                        s:fake_remove()
+                    else
+                        local cur_geo = s.geometry
+                        for _, v in ipairs {"x", "y", "width", "height" } do
+                            cur_geo[v] = geos[i][v] or cur_geo[v]
+                        end
+                        s:fake_resize(
+                            cur_geo.x,
+                            cur_geo.y,
+                            cur_geo.width,
+                            cur_geo.height
+                        )
+                    end
+                end
+
+                -- Add the remaining screens
+                for i=#old_screens + 1, #geos do
+                    local geo = geos[i]
+                    screen.fake_add(geo.x, geo.y, geo.width, geo.height)
+                end
+
+                check_tag_indexes()
+            else
+                -- Move all existing screens out of the way (to avoid temporary overlapping)
+                for s in screen do
+                    s:fake_resize(canvas_w, canvas_h, 1, 1)
```

Comment:
As they are going to be deleted anyway, it doesn't really matter much. On the plus side, we need to make sure overlapping screens don't cause any "hard" errors. I don't mean to **support** that use case, but as it happens, at least make sure it doesn't cause unrecoverable errors.

So far it looks to cause no issues, I did it on purpose.